digiroad2-viite/src/main/scala/fi/liikennevirasto/viite/package.scala

```diff
@@ -73,4 +80,30 @@ package object viite {
   def isRamp(r: BaseRoadAddress): Boolean = {
     isRamp(r.roadNumber, r.track.value)
   }
+
+  def toGeomString(geometry: Seq[Point]): String = {
+    def toBD(d: Double): String = {
+      val zeroEndRegex = """(\.0+)$""".r
+      val lastZero = """(\.[0-9])0*$""".r
+      val bd = BigDecimal(d).setScale(3, BigDecimal.RoundingMode.HALF_UP).toString
+      lastZero.replaceAllIn(zeroEndRegex.replaceFirstIn(bd, ""), { m => m.group(0) } )
+    }
+    geometry.map(p =>
+      (if (p.z != 0.0)
+        Seq(p.x, p.y, p.z)
+      else
+        Seq(p.x, p.y)).map(toBD).mkString("[", ",","]")).mkString(",")
+  }
+
+  def toGeometry(geometryString: String): Seq[Point] = {
+    def toBD(s: String): Double = {
+      BigDecimal(s).setScale(3, BigDecimal.RoundingMode.HALF_UP).toDouble
+    }
+    val pointRegex = raw"\[[^\]]*]".r
```

Comment:
Just a question, does this regex work usually when I try to do a single character I need to use \\[ instead of \[.
If it does, teach me your ways.

Reply:
[a-z] is a Regex structure meaning "any character from a to z"
the caret in [^a-z] reverses the logic and it becomes "any character that is not from a to z"
For brackets one must use the backslash to escape it from being parsed as a regex rule.